services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -219,17 +221,20 @@ const deleteIndexedBlocks = async job => {
 					const transactionsDB = await getTransactionsIndex();
 					const { transactions, assets, ...blockHeader } = block;
 
-					await BluebirdPromise.map(
+					const forkedTransactions = await BluebirdPromise.map(
 						transactions,
 						async (tx) => {
 							// Invoke 'revertTransaction' to execute command specific reverting logic
 							await revertTransaction(blockHeader, tx, dbTrx);
 
 							const forkedTransactionIDs = await getTransactionsByBlockIDs([blockHeader.id]);
```

Comment:
Fetching transactions based on the `blockID` while looping over each transaction is incorrect. This block of code needs some refactoring.